Makefile

```diff
@@ -139,3 +141,16 @@ package-pkg:
 	@tar -czf $(PKG_GZIP) --exclude='*.log' --exclude='*.string' --exclude='*.pprof' --exclude='*.gz' $(PKG_TMP)
 	@mv alibaba-pkg.gz tool/data/
 	@rm -rf $(PKG_TMP)
+
+#-------------------------------------------------------------------------------
+# Code Quality: Linting with golangci-lint
+.PHONY: lint
+LINTER := $(tool_bin)/golangci-lint
+lint:
+	@if [ ! -x "$(LINTER)" ]; then \
+  		echo "golangci-lint not found, installing to $(tool_bin)..."; \
+		mkdir -p $(tool_bin); \
+		GOBIN=$(tool_bin) go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.64.8; \
+    fi
+	@echo "Running golangci-lint..."
+	$(LINTER) run -v --config .golangci.yml ./...
```

Comment:
I suggest using go install directly, and then locating the linter executable according to the behavior of it(normally $GOPATH/bin/linter or $HOME/go/bin/linter according to https://github.com/golang/go/blob/b2960e35804aafbbb0df9973f99b034bea8c150a/src/cmd/go/internal/work/build.go#L565-L571). 

And the linter can be set as prerequisite of all build tasks. And please format it